fe/fe-core/src/main/java/com/starrocks/catalog/JDBCTable.java

```diff
@@ -180,9 +182,33 @@ private static String buildCatalogDriveName(String uri) {
         // -> jdbc_postgresql_172.26.194.237_5432_db_pg_select
         // requirement: it should be used as local path.
         // and there is no ':' in it to avoid be parsed into non-local filesystem.
-        return uri.replace("//", "")
-                .replace("/", "_")
-                .replace(":", "_");
+        String ans = uri.replaceAll("[^0-9a-zA-Z]", "_");
+
+        // currently we use this uri as part of name of download file.
+        // so if this uri is too long, we might fail to write file on BE side.
+        // so here we have to shorten it to reduce fail probability because of long file name.
+        // in most cases, jdbc uri is not that long, so we can preserve original information.
+
+        final String prefix = "jdbc_";
+        final int signLength = 64;
+        if (ans.length() > (signLength + prefix.length())) {
```

Comment:
can we not distinguish length? All driver names are generated in some way instead of using URL directly. We only need to ensure that the name generated by the same jdbc uri is the same.